### ISSUE TITLE
Parallel packet parsing causes error

### ISSUE THREAD

**Issue by samaloney** [i4Ds/STIXCore#45]
I think we could try to make the sqlite read only and also put it in memory rather than disk.

```
with ProcessPoolExecutor() as exec:
    ...:     res = exec.map(Packet, packet_data)
    ...: packets = wait(res)

---------------------------------------------------------------------------
_RemoteTraceback                          Traceback (most recent call last)
_RemoteTraceback:
"""
Traceback (most recent call last):
  File "/Users/shane/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 232, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/Users/shane/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 191, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/Users/shane/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py", line 191, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/Users/shane/Projects/STIX/STIXCore/stixcore/tmtc/packet_factory.py", line 82, in __call__
    packet = self._check_registered(sph)
  File "/Users/shane/Projects/STIX/STIXCore/stixcore/tmtc/packet_factory.py", line 60, in _check_registered
    return PacketType(data)
  File "/Users/shane/Projects/STIX/STIXCore/stixcore/tmtc/packets.py", line 238, in __init__
    self.data_header.service_subtype)
  File "/Users/shane/Projects/STIX/STIXCore/stixcore/idb/idb.py", line 671, in get_packet_pi1_val_position
    res = self._execute(sql, args, result_type='dict')
  File "/Users/shane/Projects/STIX/STIXCore/stixcore/idb/idb.py", line 578, in _execute
    self.cur.execute(sql, arguments)
sqlite3.OperationalError: disk I/O error
"""

The above exception was the direct cause of the following exception:

OperationalError                          Traceback (most recent call last)
<ipython-input-48-968ad4e21560> in <module>
      1 with ProcessPoolExecutor() as exec:
      2     res = exec.map(Packet, packet_data)
----> 3 packets = wait(res)

~/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py in wait(fs, timeout, return_when)
    282         futures.
    283     """
--> 284     with _AcquireFutures(fs):
    285         done = set(f for f in fs
    286                    if f._state in [CANCELLED_AND_NOTIFIED, FINISHED])

~/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py in __init__(self, futures)
    140
    141     def __init__(self, futures):
--> 142         self.futures = sorted(futures, key=id)
    143
    144     def __enter__(self):

~/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/process.py in _chain_from_iterable_of_lists(iterable)
    474     careful not to keep references to yielded objects.
    475     """
--> 476     for element in iterable:
    477         element.reverse()
    478         while element:

~/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py in result_iterator()
    584                     # Careful not to keep a reference to the popped future
    585                     if timeout is None:
--> 586                         yield fs.pop().result()
    587                     else:
    588                         yield fs.pop().result(end_time - time.time())

~/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py in result(self, timeout)
    423                 raise CancelledError()
    424             elif self._state == FINISHED:
--> 425                 return self.__get_result()
    426
    427             self._condition.wait(timeout)

~/.pyenv/versions/3.7.0/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

OperationalError: disk I/O error
```

**Comment by samaloney** [comment]
maybe transferring to memory would work `sqlite.connect(":memory:", check_same_thread = False)` would need to dump the data from the disabused db

**Comment by samaloney** [comment]
Fixed in #47 but I think we need to review our approach so created a new issue #48
